modules/UI/authentication/AuthHandler.js

```diff
@@ -139,47 +139,40 @@ function initJWTTokenListener(room) {
  * @param {JitsiConference} room
  * @param {string} [lockPassword] password to use if the conference is locked
  */
-function doXmppAuth (room, lockPassword) {
-    let loginDialog = LoginDialog.showAuthDialog(function (id, password) {
-        // auth "on the fly":
-        // 1. open new connection with proper id and password
-        // 2. connect to the room
-        // (this will store sessionId in the localStorage)
-        // 3. close new connection
-        // 4. reallocate focus in current room
-        openConnection({id, password, roomName: room.getName()}).then(
-        function (connection) {
-            // open room
-            let newRoom = connection.initJitsiConference(
-                room.getName(), APP.conference._getConferenceOptions()
-            );
-
-            loginDialog.displayConnectionStatus('connection.FETCH_SESSION_ID');
+function doXmppAuth(room, lockPassword) {
+    const loginDialog = LoginDialog.showAuthDialog(
+        /* successCallback */ (id, password) => {
+            const authConnection = room.createAuthenticationConnection();
 
-            newRoom.room.moderator.authenticate().then(function () {
-                connection.disconnect();
+            authConnection.authenticateAndUpgradeRole({
+                id,
+                password,
+                roomPassword: lockPassword,
 
+                /** Called when the XMPP login succeeds. */
+                onLoginSuccessful() {
+                    loginDialog.displayConnectionStatus(
+                        'connection.FETCH_SESSION_ID');
+                }
+            })
+            .then(() => {
                 loginDialog.displayConnectionStatus(
                     'connection.GOT_SESSION_ID');
```

Comment:
The equivalents of `loginDialog.displayConnectionStatus` for `connection.FETCH_SESSION_ID` and `connection.GOT_SESSION_ID` are missing in the mobile/react-native implementation which, at the time of this writing, puts mobile at a disadvantage in comparison to Web.